src/platform/packages/shared/kbn-esql-ast/src/commands_registry/commands/rerank/autocomplete.ts

```diff
@@ -19,18 +39,253 @@ export async function autocomplete(
+      return getContextualSuggestions({
+        innerText,
+        callbacks,
+        context,
+        suggestionType: 'boolean_expression',
+        expressionRoot: positionContext?.expressionRoot,
+      });
+    }
+
+    case CaretPosition.WITHIN_MAP_EXPRESSION: {
+      const endpoints = context?.inferenceEndpoints;
+      return handleMapExpression(innerText, endpoints);
+    }
+
+    case CaretPosition.AFTER_COMMAND: {
+      return [pipeCompleteItem];
+    }
+
+    default: {
+      return [];
+    }
+  }
+}
+
+// ============================================================================
+// Suggestion Handlers
+// ============================================================================
+
+async function getContextualSuggestions({
+  innerText,
+  callbacks,
+  context,
+  suggestionType,
+  expressionRoot,
+}: {
+  innerText: string;
+  callbacks?: ICommandCallbacks;
+  context?: ICommandContext;
+  suggestionType: 'field_list' | 'boolean_expression';
+  expressionRoot?: ESQLSingleAstItem;
+}): Promise<ISuggestionItem[]> {
   if (!callbacks?.getByType) {
     return [];
   }
 
-  const suggestions: ISuggestionItem[] = [];
+  switch (suggestionType) {
+    case 'field_list': {
+      const fieldSuggestions =
+        (await callbacks.getByType(['keyword', 'text', 'boolean', 'integer', 'double', 'long'])) ??
+        [];
+
+      return handleFragment(
+        innerText,
+        // check if fragment is completed
+        (fragment) => !!columnExists(fragment, context),
+        // incomplete: get available fields suggestions
+        (_: string, rangeToReplace?: { start: number; end: number }) => {
+          return fieldSuggestions.map((suggestion) => ({
+            ...suggestion,
+            text: `${suggestion.text}`,
+            rangeToReplace,
+            command: TRIGGER_SUGGESTION_COMMAND,
+          }));
+        },
+        // complete: get next actions suggestions for completed field
+        (fragment: string, rangeToReplace: { start: number; end: number }) => {
+          const suggestions = buildNextActions({ includeAssign: true, withSpaces: true });
+
+          return suggestions.map((suggestion) => ({
+            ...suggestion,
+            filterText: fragment,
+            text: fragment + suggestion.text,
+            rangeToReplace,
+            command: TRIGGER_SUGGESTION_COMMAND,
+          }));
+        }
+      );
+    }
+
+    case 'boolean_expression': {
+      if (expressionRoot) {
+        const expressionType = getExpressionType(
+          expressionRoot,
+          context?.fields,
+          context?.userDefinedColumns
+        );
+
+        if (expressionType === 'boolean' && isExpressionComplete(expressionType, innerText)) {
+          return buildNextActions({ includeBinaryOperators: true });
+        }
+      }
+
+      return await suggestForExpression({
+        innerText,
+        getColumnsByType: callbacks.getByType,
+        expressionRoot,
+        location: Location.WHERE,
```

Comment:
No definitions include RERANK as location

Reply:
You should add RERANK in the locations and rerun the script. We did the same for completion